src/quantize_int8.cpp

```diff
@@ -0,0 +1,300 @@
+#include <migraphx/float_equal.hpp>
+#include <migraphx/instruction_ref.hpp>
+#include <migraphx/quantization.hpp>
+#include <migraphx/quantize_int8.hpp>
+#include <migraphx/quantize_util.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/op/capture.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/target.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <set>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_INT8_QUANTIZATION_PARAMS)
+
+static void ins_quantize_int8(module& modl,
+                              instruction_ref ins,
+                              std::vector<instruction_ref>& converted_inputs,
+                              const std::vector<std::pair<float, float>>& ins_quant_params)
+{
+    auto orig_type = ins->get_shape().type();
+    auto inputs    = ins->inputs();
+    if(ins->name() == "dot")
+    {
+        auto dot_val = ins->get_operator().to_value();
+        assert(contains(dot_val, "alpha"));
+        assert(contains(dot_val, "beta"));
+        auto dot_alpha  = dot_val.at("alpha").to<float>();
+        auto dot_beta   = dot_val.at("beta").to<float>();
+        float scale_val = dot_alpha / (ins_quant_params[0].first * ins_quant_params[1].first);
+        float beta      = (inputs.size() == 3) ? dot_beta : 0.0f;
+        // We need additional checking about the quant_alpha value. If
+        // abs(quant_alpha) > 50 (some tmp value set here), we can convert
+        // it to an integer as the new_alpha in the quant_dot
+        instruction_ref input_c{};
+        if(converted_inputs.size() == 3)
+        {
+            input_c = converted_inputs.back();
+            converted_inputs.pop_back();
+        }
+
+        auto quant_dot = modl.insert_instruction(
+            ins, make_op("quant_dot", {{"alpha", 1}, {"beta", 0}}), converted_inputs);
```

Comment:
Why is this inserting a `quant_dot`? This should just insert the qdq pairs and then we already have a later pass to rewrite it to `quant_dot`.

Reply:
If we do what you suggested, then the ref target will never be able to run the quant_dot since ref target does not rewrite it to the quant_dot. Is that what we want?

Reply:
Then we should probably add the qdq transform to the ref. This is because verifying with int8 could be off in the driver since we dont run calibration data.

Reply:
I just added this pass in the quantization_int8() api implementation, so we will get the int8 version of the ops to be quantized.